### PR TITLE
toolchain/gcc : Add support for distro as a target

### PR DESCRIPTION
**Pull request by sacsant**
gcc test case by default downloads latest upstream code and runs the
selftests that comes with it. There is no option to use gcc tests
bundled with source pacakge available in a Linux distribution release.

Add support for distro as a target via a yaml file.

Signed-off-by: Sachin Sant <sachinp@linux.ibm.com>

### CODE REVIEW

**Comment by sacsant** [comment]
Upstream as a target

avocado run --test-runner runner toolchain/gcc.py --mux-yaml toolchain/gcc.py.data/u.yaml 
Fetching asset from toolchain/gcc.py:GCC.test
JOB ID     : 419ef709d91916c61945b9979a8d190cb1d56a47
JOB LOG    : /home/avocado/results/job-2022-04-08T01.23-419ef70/job.log
 (1/1) toolchain/gcc.py:GCC.test;run-run_type-upstream-2f21: FAIL: Few gcc tests failed,refer the log file (4498.28 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado/results/job-2022-04-08T01.23-419ef70/results.html
JOB TIME   : 4498.82 s


Distro as a target

avocado run --test-runner runner toolchain/gcc.py --mux-yaml toolchain/gcc.py.data/d.yaml 
Fetching asset from toolchain/gcc.py:GCC.test
JOB ID     : bb000ec004dc5a4177eb1d265d1ced7fc84d8aa5
JOB LOG    : /home/avocado/results/job-2022-04-08T06.37-bb000ec/job.log
 (1/1) toolchain/gcc.py:GCC.test;run-run_type-distro-8228: FAIL: Few gcc tests failed,refer the log file (2077.54 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado/results/job-2022-04-08T06.37-bb000ec/results.html
JOB TIME   : 2093.98 s

gcc test failures indicated in the above runs are related to gcc code and will be tracked separately.

**Comment by sacsant** [comment]
With certain Linux release I have observed get_source() API fails to populate source tree. This could be avocado API issue. Until it is debugged I have added an explicit fail.

File: toolchain/gcc.py [hunk 1]
```diff
@@ -46,34 +46,58 @@ def setUp(self):
...
                              'texinfo-tex', 'texinfo', 'elfutils-libelf-devel',
                              'gmp-devel', 'mpfr-devel', 'libmpc-devel',
-                             'zlib-devel', 'gettext', 'libgcc', 'libgomp'])
-        if dist.name == 'rhel' and (int(dist.version) == 8 and int(dist.release) >= 6):
-            packages.extend(['autogen', 'guile', 'guile-devel'])
+                             'zlib-devel', 'gettext', 'libgcc', 'libgomp',
+                             'dblatex', 'doxygen', 'texlive-collection-latex',
+                             'python3-sphinx', 'systemtap-sdt-devel'])
+        if dist.name == 'rhel' and \
+           (int(dist.version) == 8 and int(dist.release) >= 6):
```
Comment (PraveenPenguin) [inline toolchain/gcc.py:64]: can we move this next commit as style fix

File: toolchain/gcc.py [hunk 2]
```diff
@@ -46,34 +46,58 @@ def setUp(self):
...
+            else:
+                self.sourcedir = smm.get_source("gcc", self.sourcedir)
         os.chdir(self.sourcedir)
         process.run('./configure', ignore_status=True, sudo=True)
         build.make(self.sourcedir, ignore_status=True)
 
     def get_summary(self, index):
+        """
+        subroutine to print test result summary.
+        """
```
Comment (PraveenPenguin) [inline toolchain/gcc.py:100]: this can be clubbed with style fix

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**